pkg/operator/sync.go

```diff
@@ -97,6 +97,27 @@ func (optr *Operator) syncAll(syncFuncs []syncFunc) error {
 	return syncErr.err
 }
 
+// returns true if cloud config is required on a platform
+func isCloudConfigRequired(infra *configv1.Infrastructure) bool {
+	if infra.Spec.CloudConfig.Name != "" {
+		return true
+	}
+	for _, platform := range []configv1.PlatformType{configv1.AzurePlatformType, configv1.BareMetalPlatformType, configv1.GCPPlatformType, configv1.OpenStackPlatformType,
+		configv1.OvirtPlatformType, configv1.VSpherePlatformType} {
+		if platform == infra.Status.PlatformStatus.Type {
+			return true
+		}
+	}
+	return false
+}
+
+func processCloudConfigMapError(err error, infra *configv1.Infrastructure) error {
+	if apierrors.IsNotFound(err) && isCloudConfigRequired(infra) {
+		return fmt.Errorf("%s/%s configmap not found: %v", "openshift-config-managed", "kube-cloud-config", err)
+	}
+	return nil
```

Comment:
```suggestion
	if err != nil && ! apierrors.IsNotFound(err) {return err}
	return nil
```

that helps the caller to not worry about handling err and it can return it as-is better

Reply:
should be addressed now.